core/src/nats.rs

```diff
@@ -374,7 +374,38 @@ impl TypedNats {
             .await
             .to_anyhow()?;
 
-        let stream = consumer.messages().await.to_anyhow()?;
+        let stream: Messages = consumer.messages().await.to_anyhow()?;
+
+        Ok(JetstreamSubscription {
+            stream,
+            _ph: PhantomData::default(),
+        })
+    }
+
+    pub async fn subscribe_jetstream_ordered<T: JetStreamable>(
+        &self,
+        subject: SubscribeSubject<T>,
+    ) -> Result<JetstreamSubscription<T>> {
+        //TODO: use ordered consumers for this
```

Comment:
I was trying to use one of these: https://docs.rs/async-nats/latest/async_nats/jetstream/consumer/type.OrderedPushConsumer.html#
with a wrapper enum implementing .next(), but I had an absolutely miserable time and couldn't figure out how to get the compiler to stop complaining about lifetimes, so I've kept it this way.
another option is to have an ordered: Bool key that toggles the max_ack_pending instead of another function, wdyt @paulgb